### PR TITLE
Enable usage of ExpressionEvalContext while evaluating table function arguments

### PR DESCRIPTION
**Pull request by gierlachg**
Refactored table functions, so the evaluation of arguments is performed during `DAG` creation - it's an enabler for future work on dynamic parameters.

### CODE REVIEW

**Comment by viliam-durina** [comment]
>evaluation of arguments is performed during DAG creation - it's an enabler for future work on dynamic parameters.

That's too early for dynamic params - even though they are known during DAG creation, they can be different at runtime.

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/generator/SeriesTable.java [hunk 1]
```diff
@@ -55,32 +54,37 @@
...
         return SourceBuilder
                 .batch("series", ctx -> {
                     InternalSerializationService serializationService = ((ProcSupplierCtx) ctx).serializationService();
                     SimpleExpressionEvalContext context = new SimpleExpressionEvalContext(serializationService);
+
+                    Integer start = evaluate(argumentExpressions.get(0), null, context);
+                    Integer stop = evaluate(argumentExpressions.get(1), null, context);
+                    Integer step = evaluate(argumentExpressions.get(2), 1, context);
+                    if (start == null || stop == null || step == null) {
+                        throw QueryException.error("null arguments to GENERATE_SERIES functions");
```
Comment (viliam-durina) [inline hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/generator/SeriesTable.java:67]:
```suggestion
                        throw QueryException.error("null argument to GENERATE_SERIES function");
```

File: hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/generator/SeriesTable.java [hunk 2]
```diff
@@ -55,32 +54,37 @@
...
+                    }
+                    if (step == 0) {
+                        throw QueryException.error("step cannot equal zero");
```
Comment (viliam-durina) [inline hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/generator/SeriesTable.java:70]:
```suggestion
                        throw QueryException.error("step cannot be equal to zero");
```

File: hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/generator/SqlStreamGeneratorTest.java [hunk 3]
```diff
@@ -112,8 +112,8 @@ public void when_coercionIsRequired_then_throws() {
 
     @Test
     public void test_nullArgument() {
-        assertThatThrownBy(() -> sqlService.execute("SELECT * FROM TABLE(GENERATE_STREAM(null))"))
-                .hasMessage("rate cannot be null");
+        assertThatThrownBy(() -> sqlService.execute("SELECT * FROM TABLE(GENERATE_STREAM(null))").iterator().next())
+                .hasMessageContaining("rate cannot be null");
```
Comment (viliam-durina) [inline hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/generator/SqlStreamGeneratorTest.java:116]: Will the user know that this applies to the `GENERATE_STREAM` function? Imagine this function is in a huge query. I'd like to see it asserted by this test.

Same for other errors in this PR.

**Review by viliam-durina — APPROVED** [review]

**Review by Fly-Style — APPROVED** [review]
LGTM

**Comment by gierlachg** [comment]
run-lab-run

**Pull request merged**